java-checks/src/main/java/org/sonar/java/checks/ConcatenationWithStringValueOfCheck.java

```diff
@@ -69,13 +71,31 @@ public void visitBinaryExpression(BinaryExpressionTree tree) {
     }
 
     if (flagIssue) {
-      for (ExpressionTree valueOfTree : valueOfTrees) {
-        context.reportIssue(this, valueOfTree, "Directly append the argument of String.valueOf().");
+      for (MethodInvocationTree valueOfTree : valueOfTrees) {
+        QuickFixHelper.newIssue(context)
+          .forRule(this)
+          .onTree(valueOfTree)
+          .withMessage("Directly append the argument of String.valueOf().")
+          .withQuickFix(() -> createQuickFix(valueOfTree))
+          .report();
       }
     }
     scan(current);
   }
 
+  private JavaQuickFix createQuickFix(MethodInvocationTree invocationTree) {
+    ExpressionTree argumentTree = invocationTree.arguments().get(0);
+    String replacement = QuickFixHelper.contentForTree(argumentTree, context);
+
+    if (argumentTree instanceof BinaryExpressionTree && !argumentTree.symbolType().is("java.lang.String")) {
+      replacement = "(" + replacement + ")";
+    }
+
+    return JavaQuickFix.newQuickFix("Replace String.valueOf() with its argument")
```

Comment:
How about "Remove redundant String.valueOf wrapping"
```suggestion
    return JavaQuickFix.newQuickFix("Remove redundant String.valueOf wrapping")
```